src/window.c

```diff
@@ -6382,6 +6382,8 @@ win_fix_scroll(int resize)
 		scroll_to_fraction(wp, wp->w_prev_height);
 		wp->w_cursor.lnum = lnum;
 	    }
+	    if (wp == curwin)
+		wp->w_valid &= ~VALID_CROW;
 	    invalidate_botline_win(wp);
```

Comment:
I couldn't figure out a way to re-use validate_cursor()/check_cursor_moved(). This is the call we are skipping in
win_new_height() that is responsible for invalidating w_crow but using that unconditionally in win_fix_scroll/cursor results in scrolling. Invalidating it only seems to be necessary for curwin, and not invalidating it seems to cause an issue only in this callback case. Not sure if it makes sense to also guard behind get_callback_depth().